pkg/apis/core/v1/resource_tags.go

```diff
@@ -0,0 +1,59 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+func Tag(ctx context.Context, a api.API, obj types.IdentifiedObject, tags ...string) error {
+	return eachTagOnObject(obj, tags, func(identifier, tag string) error {
+		if err := a.Create(ctx, &ResourceWithTag{Identifier: identifier, Tag: tag}); err != nil {
+			if err, ok := err.(api.HTTPError); !ok || err.StatusCode() != 422 {
+				return fmt.Errorf("failed to tag resource: %w", err)
+			}
+		}
+		return nil
+	})
+}
+
+func Untag(ctx context.Context, a api.API, obj types.IdentifiedObject, tags ...string) error {
+	return eachTagOnObject(obj, tags, func(identifier, tag string) error {
+		if err := a.Destroy(ctx, &ResourceWithTag{Identifier: identifier, Tag: tag}); api.IgnoreNotFound(err) != nil {
+			return fmt.Errorf("failed to untag resource: %w", err)
+		}
+		return nil
+	})
+}
+
+func eachTagOnObject(obj types.IdentifiedObject, tags []string, callback func(identifier, tag string) error) error {
```

Comment:
refactor this to return an array of objects to create/destroy and loop in `Tag`/`Untag`